modulemd/v2/tests/test-modulemd-modulestream.c

```diff
@@ -0,0 +1,164 @@
+  /* Test with a zero mdversion */
+  stream = modulemd_module_stream_new (
+    MD_MODULESTREAM_VERSION_LATEST + 1, "foo", "latest");
+  g_assert_null (stream);
+  g_clear_object (&stream);
+}
+
+static void
+module_stream_test_copy (ModuleStreamFixture *fixture, gconstpointer user_data)
+{
+  g_autoptr (ModulemdModuleStream) stream = NULL;
+  g_autoptr (ModulemdModuleStream) copied_stream = NULL;
+  guint64 version;
+
+  for (version = MD_MODULESTREAM_VERSION_ONE;
+       version <= MD_MODULESTREAM_VERSION_LATEST;
+       version++)
+    {
+      /* Test copying with a stream name */
+      stream = modulemd_module_stream_new (version, "foo", "latest");
+      copied_stream = modulemd_module_stream_copy (stream, NULL);
+      g_assert_nonnull (copied_stream);
+      g_assert_true (MODULEMD_IS_MODULE_STREAM (copied_stream));
+      g_assert_cmpstr (modulemd_module_stream_get_module_name (stream),
+                       ==,
+                       modulemd_module_stream_get_module_name (copied_stream));
+      g_assert_cmpstr (modulemd_module_stream_get_stream_name (stream),
+                       ==,
+                       modulemd_module_stream_get_stream_name (copied_stream));
+      g_clear_object (&stream);
+      g_clear_object (&copied_stream);
+
+
+      /* Test copying without a stream name */
+      stream = modulemd_module_stream_new (version, "foo", NULL);
+      copied_stream = modulemd_module_stream_copy (stream, NULL);
+      g_assert_nonnull (copied_stream);
+      g_assert_true (MODULEMD_IS_MODULE_STREAM (copied_stream));
+      g_assert_cmpstr (modulemd_module_stream_get_module_name (stream),
+                       ==,
+                       modulemd_module_stream_get_module_name (copied_stream));
+      g_assert_cmpstr (modulemd_module_stream_get_stream_name (stream),
+                       ==,
+                       modulemd_module_stream_get_stream_name (copied_stream));
+      g_clear_object (&stream);
+      g_clear_object (&copied_stream);
+
+      /* Test copying with and renaming the stream name */
+      stream = modulemd_module_stream_new (version, "foo", "latest");
+      copied_stream = modulemd_module_stream_copy (stream, "earliest");
+      g_assert_nonnull (copied_stream);
+      g_assert_true (MODULEMD_IS_MODULE_STREAM (copied_stream));
+      g_assert_cmpstr (modulemd_module_stream_get_module_name (stream),
+                       ==,
+                       modulemd_module_stream_get_module_name (copied_stream));
+      g_assert_cmpstr (
+        modulemd_module_stream_get_stream_name (stream), ==, "latest");
+      g_assert_cmpstr (modulemd_module_stream_get_stream_name (copied_stream),
+                       ==,
+                       "earliest");
+      g_clear_object (&stream);
+      g_clear_object (&copied_stream);
+    }
+}
+
+int
+main (int argc, char *argv[])
+{
+  setlocale (LC_ALL, "");
+  g_test_init (&argc, &argv, NULL);
+  g_test_bug_base ("https://bugzilla.redhat.com/show_bug.cgi?id=");
```

Comment:
nope ;)

Reply:
Could you leave a more productive comment, @ignatenkobrain ? What part do you disagree with?